### ISSUE TITLE
Long tab names styling

### ISSUE THREAD

**Issue by ahetawal-p** [react-native-community/react-native-tab-view#284]
Hello,
Am trying to find if there is a way add `ellipsis` to the long label names. 
Currently the names get wrapped to a second line, which doesnt looks good.

![screen shot 2017-06-22 at 11 48 44 am](https://user-images.githubusercontent.com/8919529/27450574-f3410bb2-5740-11e7-926c-acc9fe5fea48.png)

Any ideas how can i get them to `nowrap` ?

**Comment by MarcoDriessen** [comment]
You can return your own Text component where you can add ellipsizeMode to. With renderLabel in the TabBarTop component. For instance:

```
            <TabBarTop
                {...props}
                renderLabel={this.renderLabel}
            />
```
```
    renderLabel = () => {
        return (
            <Text
                style={styles.label}
                numberOfLines={1}
                ellipsizeMode='tail'
            >
               This is a long text you see
            </Text>
        );
    };
```
This has worked for me.

**Comment by satya164** [comment]
If you need to hide a part of the name of the tab, you should probably think of a shorter name. It's more difficult to recognize the tab for the user if you don't show the full name.

Material design guidelines also advise against it: https://material.io/guidelines/components/tabs.html#tabs-usage

**Comment by navata** [comment]
I have same issue. I think we need have property for width auto text. The default width will same as each tab. Admin should update this function in our library:
```
_getTabWidth = props => {
    const { layout, navigationState, tabStyle } = props;
    const flattened = StyleSheet.flatten(tabStyle);

    if (flattened) {
      switch (typeof flattened.width) {
        case 'number':
          return flattened.width;
        case 'string':
          if (flattened.width.endsWith('%')) {
            const width = parseFloat(flattened.width);
            if (Number.isFinite(width)) {
              return layout.width * (width / 100);
            }
          }
      }
    }

    if (props.scrollEnabled) {
      return (layout.width / 5) * 2;
    }

    return layout.width / navigationState.routes.length;
  };
```

**Comment by Kielan** [comment]
An api option for this component to simulate a whitespace: 'no wrap' feature as exists in web should be available for this component. I would be happy to submit a pull request.

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
